### PR TITLE
Fix ingesting tarballs with leading slashes

### PR DESCRIPTION
**Pull request by siscia**
Fix the ingestion of tarballs when the file inside the tarball present leading slashes.

### CODE REVIEW

File: cvmfs/sync_union_tarball.cc [hunk 1]
```diff
@@ -276,6 +276,13 @@ std::string SyncUnionTarball::SanitizePath(const std::string &path) {
       return to_return;
     }
   }
+  if (path.length() >= 1) {
+    if (path[0] == '/') {
+      std::string to_return(path);
+      to_return.erase(0, 1);
```
Comment (jblomer) [inline cvmfs/sync_union_tarball.cc]: Side note, why not simply return `path.substr(1)`.

**Review by jblomer — CHANGES_REQUESTED** [review]
Looks in principal good to me, just one suggestion for an improvement.

**Comment by jblomer** [comment]
Please also add a PR for the backport to the 2.7 branch

**Review by jblomer — APPROVED** [review]

**Comment by siscia** [comment]
Still something isn't working quite alright in the test machine... Looking into it.

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Pull request merged**